bowtie/_cli.py

```diff
@@ -767,6 +767,21 @@ def _check_fail_fast_provided(
     return value
 
 
+def _validate_schema_and_dialect(
+    schema: Any,
+    dialect: Any,
+) -> bool:
+
+    if "$schema" in schema:
```

Comment:
Your change does very similar things in 3 places, when I'd expect it to only do it in one -- specifically notice how you check `if "$schema" in schema` in all 3 places you're changing. You'll find that's unnecessary in at least one place.

A good thing for you to do might now be to completely remove your code changes and leave only your test changes, and then add the absolute minimum change required to get your test to pass, which should reveal how to do this in a shorter way.